dsindex/indexer.go

```diff
@@ -7,103 +7,105 @@ import (
-	Delete(ctx context.Context, index, key string) error
+	Delete(ctx context.Context, key, value string) error
 
-	// DeleteIndex deletes all keys in the given index.  If an index is not in
-	// the datastore, this method returns no error.
-	DeleteIndex(ctx context.Context, index string) (count int, err error)
+	// DeleteKey deletes all values in the given key.  If a key is not in the
+	// datastore, this method returns no error.  Returns a count of values that
+	// were deleted.
+	DeleteKey(ctx context.Context, key string) (count int, err error)
 
-	// DeleteAll deletes all indexes managed by this Indexer
+	// DeleteAll deletes all keys managed by this Indexer.  Returns a count of
+	// the values that were deleted.
 	DeleteAll(ctx context.Context) (count int, err error)
 
-	// ForEach calls the function for each key in the specified index, until
-	// there are no more keys, or until the function returns false.  If index
-	// is empty string, then all indexs are iterated.
-	ForEach(ctx context.Context, index string, fn func(index, key string) bool) error
+	// ForEach calls the function for each value in the specified index, until
+	// there are no more values, or until the function returns false.  If key
+	// is empty string, then all keys are iterated.
+	ForEach(ctx context.Context, index string, fn func(key, value string) bool) error
 
-	// HasKey determines if the index contains the specified key
-	HasKey(ctx context.Context, index, key string) (bool, error)
+	// HasValue determines if the key contains the specified value
+	HasValue(ctx context.Context, key, value string) (bool, error)
 
-	// HasAny determines if any key is in the specified index.  If index is
-	// empty string, then all indexes are searched.
-	HasAny(ctx context.Context, index string) (bool, error)
+	// HasAny determines if any value is in the specified key.  If key is
+	// empty string, then all values are searched.
+	HasAny(ctx context.Context, key string) (bool, error)
 
-	// Search returns all keys for the given index
-	Search(ctx context.Context, index string) (ids []string, err error)
+	// Search returns all values for the given key
+	Search(ctx context.Context, key string) (values []string, err error)
 }
 
 // indexer is a simple implementation of Indexer.  This implementation relies
-// on the underlying data store supporting efficent querying by prefix.
+// on the underlying data store to support efficient querying by prefix.
 //
 // TODO: Consider adding caching
 type indexer struct {
-	dstore    ds.Datastore
-	indexPath string
+	dstore ds.Datastore
 }
 
-// New creates a new datastore index.  All indexes are stored prefixed with the
-// specified index path.
+// New creates a new datastore index.  All indexes are stored under the
+// specified index name.
 //
 // To persist the actions of calling Indexer functions, it is necessary to call
 // dstore.Sync.
-func New(dstore ds.Datastore, indexPath string) Indexer {
+func New(dstore ds.Datastore, name string) Indexer {
```

Comment:
given that we're turning name into ds.Key anyway any reason not to just pass it in?

Reply:
Done.

True. Nicer to pass a type here.